subprojects/entitled-mirror/README.md

```diff
@@ -0,0 +1,76 @@
+# Entitled Mirror
+This directory contains Containerfiles and an OpenShift template for deploying an entitled package repository mirror (using yum `reposync` and NGINX) to facilitate driver compilation on CI clusters.
+
+## Overview
+The mirror is expected to only be deployed once and run indefinitely for use by CI clusters.
+
+## Prerequisites
+- An OpenShift cluster
+- The github.com/omertuc forked openshift-acme controller (see `./deploy_acme.sh`) must be running on the cluster
+- At least one entitled (SKU MCT2741F3) Node with label `entitled="true"`
+- A CA certificate which you can generate client certificates from. Follow `./auth/README.md` for more information, scripts and examples.
+
+## Containers
+The deployment makes use of two container images:
+- The `sync` init container, hosted in `quay.io/openshift-psap/entitled-mirror:sync`. This container is responsible for running `reposync` for all the mirrored repositories to make sure they're ready for mirroring.
+- The `serve` container, hosted in `quay.io/openshift-psap/entitled-mirror:serve`. This container is responsible for running the NGINX server to serve the contents synced by the init container.
+
+If you wish to build & push new versions of the containers images, `podman login` to your quay.io account (which must be authorized to push to the `openshift-psap` organization) and run `./containers/build.sh` to build and `./containers/upload.sh` to push.
+
+## TLS mutual authentication
+Typically TLS (HTTPS) is only deployed to authenticate the server, but in our mirror deployment we also authenticate the client because we don't want the mirror content to be publicly available. The `serve` container is (see "Containers" section) configured to perform both authentications.
+- Server authentication certificate generation is handled by the acme controller (see "Prerequisites") using LetsEncrypt.
+- Client authentication certificate generation is handled by the `auth` directory inside this directory, see the README.md inside of it for more information. 
+
+## Storage
+The OpenShift template `openshift/template.yaml` defines a persistent volume claim for enough storage to store all the mirror packages. It makes use of the cluster's default StorageClass.
+
+# Installation
+## Initial deployment
+- Make sure your KUBECONFIG points to the correct cluster on which you wish to deploy the mirror
+- Run `./deploy.sh`
```

Comment:
before the deploy step, could you clarify where the list of repos that will be sync is defined?
I could have avoided downloaded > 15000 packages in my cluster :# :# :# 🙃

Reply:
Sure.

Although I'm not sure which repos in that list are redundant. Not sure if the OCP ones are even needed.